### PR TITLE
[torchgen] Move Executorch codegen logic into torchgen

### PR DESCRIPTION
**Pull request by larryliu0820**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89596
* __->__ #90806
* #90099
* #90098

## Codegen entry point

Main logic and Executorch codegen entry: `gen_executorch.py`.

`RegisterCodegenUnboxedKernels.cpp`:
```cpp
register_operators({
	Operator(
		"aten::add.out",
		[](EValue** stack) {
			EValue& self = *stack[0];
			EValue& other = *stack[1];
			EValue& alpha = *stack[2];
			EValue& out = *stack[3];

			const at::Tensor & self_base = self.to<at::Tensor>();
			const at::Tensor & other_base = other.to<at::Tensor>();
			const at::Scalar & alpha_base = alpha.to<at::Scalar>();
			at::Tensor & out_base = out.to<at::Tensor>();

			EXECUTORCH_SCOPE_PROF("native_call_add.out");
			torch::executor::aten::add_outf(self_base, other_base, alpha_base, out_base);
	})
);
```

`Functions.h`:
```cpp

namespace torch {
namespace executor {

namespace aten {

// aten::add_outf(Tensor self, Tensor other, Scalar alpha, *, Tensor(a!) out) -> Tensor(a!)
TORCH_API inline at::Tensor & add_outf(const at::Tensor & self, const at::Tensor & other, at::Scalar alpha, at::Tensor & out) {
    return at::add_outf(self, other, alpha, out);
}

} // namespace aten

} // namespace executor
} // namespace torch
```

* Unit tests: `test_executorch_gen.py`

CI job in next PR.

### CODE REVIEW

**Comment by larryliu0820** [comment]
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/).

File: tools/test/test_executorch_gen.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                """
+- op: add.out
+  device_check: NoCheck   # TensorIterator
+  dispatch:
+    CPU: torch::executor::add_out_kernel
+
+- op: mul.out
+  device_check: NoCheck   # TensorIterator
+  dispatch:
+    CPU: torch::executor::mul_out_kernel
```
Comment (ezyang) [inline tools/test/test_executorch_gen.py:102]: Would happily delete this test once the full Executorch op yaml goes live. Don't want to have to be updating this test code.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**